torch/_jit_internal.py

```diff
@@ -357,3 +358,36 @@ def __getitem__(self, types):
 BroadcastingList1 = BroadcastingListCls()
 for i in range(2, 7):
     globals()["BroadcastingList{}".format(i)] = BroadcastingList1
+
+# Retrieves a fully-qualified name (module hierarchy + classname) for a given obj.
+def _qualified_name(obj):
+    # short-circuit in cases where the object already has a known qualified name
+    if isinstance(obj, torch._C.Function):
+        return obj.qualified_name
+
+    name = obj.__name__
+    module_name = obj.__module__
+
+    # The Python docs are very clear that `__module__` can be None, but I can't
+    # figure out when it actually would be.
+    if module_name is None:
+        raise RuntimeError("Could not get qualified name for class '{}': "
+                           "__module__ can't be None.".format(name))
+
+    # if getattr(sys.modules[module_name], name) is not obj:
+    #     raise RuntimeError("Could not get qualified name for class '{}': "
+    #                        "the attr {} on module {} is not the the class".format(name, name, module_name))
+
+    # __main__ is a builtin module, so rewrite it to "__torch__".
+    if module_name == "__main__":
```

Comment:
Why do we do this? How can the names collide if this is all happening in the compiler? This would make pickle files with classes in them un-readable from Python without a custom unpickler to translate `__torch__` to `__main__`.

Reply:
I don't know, this is just a move of _qualname so it is accessible in attributes.py

Reply:
From @suo, it was originally for some namespace uniqueness issues but can probably be deleted, we should do that before the 1.2 release